src/Guzzle/SubjectRewritingMiddleware.php

```diff
@@ -191,9 +191,10 @@ private function updateSearchSubjects(array $subjects) : array
                 foreach ($subjects as $i => $subject) {
                     if ($replacement['id'] === $subject['id']) {
                         $subjects[$i]['results'] += $total;
-                        break;
+                        continue 2;
```

Comment:
with 3 nested loops `continue` and `break` become similar to `goto`s, so can `$this->upsertNewSubject($subjects, $replacement)` be extracted?